### PR TITLE
Feature/titan api schema

### PR DESCRIPTION
**Pull request by WernerVdM97**
Summary of this change:
Added schema validation to the outgoing inspection request of the TITAN API.
As well as fixed a bug where the custom country name, instead of the ISO code, was sent through to TITAN.

How to test this code via the UI:
Run TITAN mock API and request inspection for any given consignment that already had a previous request sent.
Compare the JSON requests before and after this update to ensure they are still the same.

### CODE REVIEW

File: lib/finished_goods/repositories/titan_repo.rb [hunk 1]
```diff
@@ -212,7 +212,7 @@ def titan_pallet_header(pallet_id, pallet, govt_inspection_sheet, destination_re
         targetCountry: destination_country || govt_inspection_sheet.iso_country_code || EcertRepo.new.find_first_country_iso_for(ecert_agreement_id),
         protocolExceptionIndicator: titan_protocol_exception || govt_inspection_sheet.titan_protocol_exception || 'NA',
         agreementCode: ecert_agreement_code,
-        upn: pallet.upn }
+        upn: pallet.upn || '' }
```
Comment (jcsjcs) [inline lib/finished_goods/repositories/titan_repo.rb:215]: Is this necessary? You seem to do this again in the validation method.

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/finished_goods/repositories/titan_repo.rb (hunk 1 above)
Comment (WernerVdM97) [reply lib/finished_goods/repositories/titan_repo.rb:215]: In the case of `validate_titan_inspection_api_payload` no it is not necessary, however, I added that to have this function match the convention of the function `titan_pallet_sequence:232` where `phytoData` is made an empty string if nil.

Comment (jcsjcs) [reply lib/finished_goods/repositories/titan_repo.rb:215]: OK. But aren't these methods only used to build up a hash for input to the dry validation?

In which case whether you pass in `''` or `nil` as a value, the output from the validation will always be `nil`.
And in that case, isn't `pre-validation-value = value || ''` just noise?

**Review by jcsjcs — CHANGES_REQUESTED** [review]

File: lib/finished_goods/repositories/titan_repo.rb (hunk 1 above)
Comment (WernerVdM97) [reply lib/finished_goods/repositories/titan_repo.rb:215]: Yes, it is just noise. I'll remove both cases then.

**Pull request merged**